documentation/quarkus/guide-beanmapping-quarkus.asciidoc

```diff
@@ -0,0 +1,157 @@
+:toc: macro
+toc::[]
+
+= Bean mapping with Quarkus
+
+This guide will show bean mapping in particular for a Quarkus application. We recommend using https://mapstruct.org/[MapStruct] with a Quarkus application because the other bean mapper frameworks are using Java reflections. They are not supported in GraalVm right now and causes problems building native applications. MapStruct is a code generator that greatly simplifies the implementation of mappings between Java bean types based on a convention over configuration approach. The mapping code will be generated at compile-time and uses plain method invocations and thus is fast, type-safe, and easy to understand. MapStruct has to be configured to not use Java reflections but it will be shown in this guide.
+
+You can find the official
+https://mapstruct.org/documentation/stable/reference/pdf/mapstruct-reference-guide.pdf[MapStruct reference guide] and a general introduction to MapStruct from https://www.baeldung.com/mapstruct[Baeldung].
+
+== MapStruct Dependency 
+To get access to MapStruct we have to add the dependency to our POM.xml:
+
+[source, xml]
+----
+<dependency>
+  <groupId>org.mapstruct</groupId>
+  <artifactId>mapstruct</artifactId>
+  <version>1.4.2.Final</version>
+  <scope>provided</scope>
+</dependency>
+----
+
+MapStruct provides an annotation processor that also has to be added to the POM.xml 
+[source, xml]
+----
+<plugin>
+    <groupId>org.apache.maven.plugins</groupId>
```

Comment:
Check the indentations in the code snippets. Sometimes use indentations with 2 spaces and sometimes with 4 spaces.